mmrotate/datasets/dota.py

```diff
@@ -1,361 +1,167 @@
                    }  # in mmdet v2.0 label is 0-based
-        ann_files = glob.glob(ann_folder + '/*.txt')
-        data_infos = []
-        if not ann_files:  # test phase
-            ann_files = glob.glob(ann_folder + '/*.png')
-            for ann_file in ann_files:
+        data_list = []
+        if self.ann_file == '':
+            img_files = glob.glob(self.data_prefix['img_path'] + '/*.png')
+            for img_path in img_files:
                 data_info = {}
-                img_id = osp.split(ann_file)[1][:-4]
+                img_id = osp.split(self.ann_file)[1][:-4]
                 img_name = img_id + '.png'
-                data_info['filename'] = img_name
-                data_info['ann'] = {}
-                data_info['ann']['bboxes'] = []
-                data_info['ann']['labels'] = []
-                data_infos.append(data_info)
+                data_info['file_name'] = img_name
+                data_info['img_path'] = img_path
+                data_info['img_id'] = osp.split(img_path)[1]
+
+                img_bytes = self.file_client.get(img_path)
+                img = mmcv.imfrombytes(img_bytes, backend='cv2')
+                width, height = img.shape[:2]
+                del img, img_bytes
+                data_info['height'] = height
+                data_info['width'] = width
+
+                instances = []
+                instance = {}
+                instance['bbox'] = []
+                instance['bbox_label'] = []
+                instance['ignore_flag'] = 0
+                instances.append(instance)
+                data_info['instances'] = instances
+                data_list.append(data_info)
+
+            return data_list
         else:
-            for ann_file in ann_files:
+            txt_files = glob.glob(self.ann_file + '/*.txt')
+            if len(txt_files) == 0:
+                raise ValueError('There is no txt file in ' f'{self.ann_file}')
+            for txt_file in txt_files:
                 data_info = {}
-                img_id = osp.split(ann_file)[1][:-4]
+                img_id = osp.split(txt_file)[1][:-4]
+                data_info['img_id'] = img_id
                 img_name = img_id + '.png'
-                data_info['filename'] = img_name
-                data_info['ann'] = {}
-                gt_bboxes = []
-                gt_labels = []
-                gt_polygons = []
-                gt_bboxes_ignore = []
-                gt_labels_ignore = []
-                gt_polygons_ignore = []
-
-                if os.path.getsize(ann_file) == 0 and self.filter_empty_gt:
-                    continue
-
-                with open(ann_file) as f:
+                data_info['file_name'] = img_name
+                data_info['img_path'] = osp.join(self.data_prefix['img_path'],
+                                                 img_name)
+                img_bytes = self.file_client.get(data_info['img_path'])
+                img = mmcv.imfrombytes(img_bytes, backend='cv2')
```

Comment:
Loading images to get width and height every time during init is not efficient. Consider providing a stript to dump image metas and read it here, as a choice.

Reply:
I will remove the `width` and `height`, just like mmrotate-v0.x. Because we don't need to filter data according to the image's shape.